engine/protocol/api.go

```diff
@@ -0,0 +1,157 @@
+package protocol
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+type API interface {
+	GetLatestBlockHeader(ctx context.Context, isSealed bool) (*flow.Header, error)
+	GetBlockHeaderByID(ctx context.Context, id flow.Identifier) (*flow.Header, error)
+	GetBlockHeaderByHeight(ctx context.Context, height uint64) (*flow.Header, error)
+	GetLatestBlock(ctx context.Context, isSealed bool) (*flow.Block, error)
+	GetBlockByID(ctx context.Context, id flow.Identifier) (*flow.Block, error)
+	GetBlockByHeight(ctx context.Context, height uint64) (*flow.Block, error)
+	GetExecutionResultByID(ctx context.Context, id flow.Identifier) (*flow.ExecutionResult, error)
```

Comment:
GetBlockByHeight, GetExecutionResultByID should go to the execution API instead. This was confirmed by Dapper.

Reply:
Updated here: https://github.com/onflow/flow-go/pull/2701